module/solutions/helper.js

```diff
@@ -2727,121 +2815,171 @@ module.exports = class SolutionsHelper {
 					}
 				}
 
-				requestedData['filter'] = {}
-				// if (solutionIds.length > 0) {
-				// 	requestedData['filter']['skipSolutions'] = solutionIds
-				// }
+				// When targetedSolutions is not empty alter the response based on the value of currentScopeOnly
+				if (targetedSolutions.data.data && targetedSolutions.data.data.length > 0) {
+					let filteredTargetedSolutions = []
+					targetedSolutions.data.data.forEach((solution) => {
+						let newEntry = Object.assign(
+							{
+								_id: '',
+								solutionId: solution._id,
+								creator: solution.creator || '',
+								solutionMetaInformation: solution.metaInformation || {},
+							},
+							_.pick(solution, [
+								'type',
+								'externalId',
+								'projectTemplateId',
+								'certificateTemplateId',
+								'programId',
+								'programName',
+								'name',
+								'description',
+							])
+						)
+						filteredTargetedSolutions.push(newEntry)
+					})
+					if (currentScopeOnly) {
+						filteredTargetedSolutions.forEach((solution) => {
+							// Find the corresponding project in mergedData where solutionId matches _id
+							const matchingProject = _.find(mergedData, (project) => {
+								return String(project.solutionId) === String(solution.solutionId)
+							})
 
-				if (filter && filter !== '') {
-					if (filter === CONSTANTS.common.CREATED_BY_ME) {
-						requestedData['filter']['isAPrivateProgram'] = {
-							$ne: false,
-						}
-					} else if (filter === CONSTANTS.common.ASSIGN_TO_ME) {
-						requestedData['filter']['isAPrivateProgram'] = false
-					}
-				}
+							if (matchingProject) {
+								// Add all keys from the matching project to the solution object
+								Object.assign(solution, matchingProject)
+							}
+						})
+						mergedData = filteredTargetedSolutions
+						totalCount = mergedData.length
+					} else if (process.env.SUBMISSION_LEVEL === 'USER') {
+						filteredTargetedSolutions.forEach((solution) => {
+							// Check if the solution _id exists in mergedData solutionId
+							const existsInMergedData = _.some(mergedData, (project) => {
+								return String(project.solutionId) === String(solution.solutionId)
+							})
 
-				let targetedSolutions = {
-					success: false,
-				}
+							if (!existsInMergedData) {
+								mergedData.push(solution)
+							}
+						})
 
-				surveyReportPage = UTILS.convertStringToBoolean(surveyReportPage)
+						totalCount = mergedData.length
+					}
+				}
 
-				let getTargetedSolution = true
+				if (search == undefined || search == '') search = ''
+				searchQuery = [{ title: new RegExp(search, 'i') }, { description: new RegExp(search, 'i') }]
+				if (process.env.SUBMISSION_LEVEL == 'ENTITY' && requestedData.hasOwnProperty('entityId')) {
+					mergedData = []
+					totalCount = 0
+					// Fetch projects accessible by the user, when acl.visibility = ALL
+					aclVisibilityAllProjects = await projectQueries.projectDocument(
+						{
```

Comment:
@Prajwal17Tunerlabs  can we do this and line 2903 n a single call. using neted queries?please check

Reply:
multiple queries clubbed together to form a single query.